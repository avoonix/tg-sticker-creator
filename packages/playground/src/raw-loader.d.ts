declare module "!!raw-loader!*" {
  declare const exp: string;
  export default exp;
}
