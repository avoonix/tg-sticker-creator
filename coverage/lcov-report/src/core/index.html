
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/core</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/core</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.3% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>435/516</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">81.13% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>86/106</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.1% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>86/113</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.98% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>402/473</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="LayerTransform.ts"><a href="LayerTransform.ts.html">LayerTransform.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="13" class="abs high">13/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	</tr>

<tr>
	<td class="file medium" data-value="Lottie.ts"><a href="Lottie.ts.html">Lottie.ts</a></td>
	<td data-value="68.91" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 68%"></div><div class="cover-empty" style="width: 32%"></div></div>
	</td>
	<td data-value="68.91" class="pct medium">68.91%</td>
	<td data-value="74" class="abs medium">51/74</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="9" class="abs low">3/9</td>
	<td data-value="55" class="pct medium">55%</td>
	<td data-value="20" class="abs medium">11/20</td>
	<td data-value="71.01" class="pct medium">71.01%</td>
	<td data-value="69" class="abs medium">49/69</td>
	</tr>

<tr>
	<td class="file high" data-value="LottieItem.ts"><a href="LottieItem.ts.html">LottieItem.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	</tr>

<tr>
	<td class="file low" data-value="Mask.ts"><a href="Mask.ts.html">Mask.ts</a></td>
	<td data-value="7.4" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 7%"></div><div class="cover-empty" style="width: 93%"></div></div>
	</td>
	<td data-value="7.4" class="pct low">7.4%</td>
	<td data-value="27" class="abs low">2/27</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="7.4" class="pct low">7.4%</td>
	<td data-value="27" class="abs low">2/27</td>
	</tr>

<tr>
	<td class="file high" data-value="Resolvable.ts"><a href="Resolvable.ts.html">Resolvable.ts</a></td>
	<td data-value="96.42" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 96%"></div><div class="cover-empty" style="width: 4%"></div></div>
	</td>
	<td data-value="96.42" class="pct high">96.42%</td>
	<td data-value="28" class="abs high">27/28</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="96" class="pct high">96%</td>
	<td data-value="25" class="abs high">24/25</td>
	</tr>

<tr>
	<td class="file medium" data-value="TransformMixin.ts"><a href="TransformMixin.ts.html">TransformMixin.ts</a></td>
	<td data-value="56.66" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 56%"></div><div class="cover-empty" style="width: 44%"></div></div>
	</td>
	<td data-value="56.66" class="pct medium">56.66%</td>
	<td data-value="30" class="abs medium">17/30</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="8" class="abs low">2/8</td>
	<td data-value="56.66" class="pct medium">56.66%</td>
	<td data-value="30" class="abs medium">17/30</td>
	</tr>

<tr>
	<td class="file high" data-value="classes.ts"><a href="classes.ts.html">classes.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file high" data-value="constants.ts"><a href="constants.ts.html">constants.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="79" class="abs high">79/79</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="24" class="abs high">24/24</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="79" class="abs high">79/79</td>
	</tr>

<tr>
	<td class="file high" data-value="internal.ts"><a href="internal.ts.html">internal.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="60" class="abs high">60/60</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="60" class="abs high">60/60</td>
	</tr>

<tr>
	<td class="file high" data-value="matchKeys.ts"><a href="matchKeys.ts.html">matchKeys.ts</a></td>
	<td data-value="95" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 95%"></div><div class="cover-empty" style="width: 5%"></div></div>
	</td>
	<td data-value="95" class="pct high">95%</td>
	<td data-value="60" class="abs high">57/60</td>
	<td data-value="89.47" class="pct high">89.47%</td>
	<td data-value="38" class="abs high">34/38</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="94.44" class="pct high">94.44%</td>
	<td data-value="54" class="abs high">51/54</td>
	</tr>

<tr>
	<td class="file high" data-value="util.ts"><a href="util.ts.html">util.ts</a></td>
	<td data-value="87.87" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 87%"></div><div class="cover-empty" style="width: 13%"></div></div>
	</td>
	<td data-value="87.87" class="pct high">87.87%</td>
	<td data-value="132" class="abs high">116/132</td>
	<td data-value="59.09" class="pct medium">59.09%</td>
	<td data-value="22" class="abs medium">13/22</td>
	<td data-value="93.02" class="pct high">93.02%</td>
	<td data-value="43" class="abs high">40/43</td>
	<td data-value="91.42" class="pct high">91.42%</td>
	<td data-value="105" class="abs high">96/105</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2022-09-06T19:07:28.057Z
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    