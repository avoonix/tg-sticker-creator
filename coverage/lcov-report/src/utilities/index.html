
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/utilities</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/utilities</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">52.3% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>34/65</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">15.38% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>2/13</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">51.61% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>16/31</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">51.61% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>32/62</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="create.ts"><a href="create.ts.html">create.ts</a></td>
	<td data-value="47.45" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 47%"></div><div class="cover-empty" style="width: 53%"></div></div>
	</td>
	<td data-value="47.45" class="pct low">47.45%</td>
	<td data-value="59" class="abs low">28/59</td>
	<td data-value="15.38" class="pct low">15.38%</td>
	<td data-value="13" class="abs low">2/13</td>
	<td data-value="51.61" class="pct medium">51.61%</td>
	<td data-value="31" class="abs medium">16/31</td>
	<td data-value="46.42" class="pct low">46.42%</td>
	<td data-value="56" class="abs low">26/56</td>
	</tr>

<tr>
	<td class="file high" data-value="enum.ts"><a href="enum.ts.html">enum.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Sat Apr 16 2022 09:52:27 GMT+0000 (Coordinated Universal Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    