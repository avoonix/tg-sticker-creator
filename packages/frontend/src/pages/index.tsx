import { authAtom } from "@/modules/export/auth";
import { getSettings } from "@/modules/export/requests";
import VideoCardList from "@/modules/overview/VideoCardList";
import { VideoEntry } from "@/modules/overview/VideoEntry";
import { Heading, View } from "@adobe/react-spectrum";
import { useAtom } from "jotai";
import Head from "next/head";
import { useEffect, useState } from "react";

const videos: VideoEntry[] = [
  {
    url: "https://tg-sticker-bot.vercel.app/api/dev/headpats/b28b1f41-10aa-4e37-a0c3-3bcca310cbe5/video.webm",
    emojis: [
      { name: "waving", encoded: "👋" },
      { name: "love", encoded: "🥰" },
      { name: "hearts", encoded: "💕" },
    ],
    name: "Headpats",
    stickerId: "headpats",
    settingId: "b28b1f41-10aa-4e37-a0c3-3bcca310cbe5",
  },
  {
    url: "https://tg-sticker-bot.vercel.app/api/dev/popping/b28b1f41-10aa-4e37-a0c3-3bcca310cbe5/video.webm",
    emojis: [
      { name: "cat", encoded: "🐱" },
      { name: "open mouth", encoded: "😮" },
      { name: "screaming cat", encoded: "🙀" },
      { name: "popping cork", encoded: "🍾" },
      { name: "bubbles", encoded: "🫧" },
    ],
    name: "Popping",
    stickerId: "popping",
    settingId: "b28b1f41-10aa-4e37-a0c3-3bcca310cbe5",
  },
  {
    url: "https://tg-sticker-bot.vercel.app/api/dev/halloween/b28b1f41-10aa-4e37-a0c3-3bcca310cbe5/video.webm",
    emojis: [{ name: "eyes", encoded: "👀" }],
    name: "Halloween",
    stickerId: "halloween",
    settingId: "b28b1f41-10aa-4e37-a0c3-3bcca310cbe5",
  },
  {
    url: "https://tg-sticker-bot.vercel.app/api/dev/attribution/b28b1f41-10aa-4e37-a0c3-3bcca310cbe5/video.webm",
    emojis: [
      { name: "info", encoded: "ℹ️" },
      { name: "writing", encoded: "✍️" },
    ],
    name: "Attribution",
    stickerId: "attribution",
    settingId: "b28b1f41-10aa-4e37-a0c3-3bcca310cbe5",
  },
];

export default function Home() {
  const [auth] = useAtom(authAtom);
  const [savedEntries, setSavedEntries] = useState<VideoEntry[]>([]);

  useEffect(() => {
    if (!auth.data || !auth.type) {
      console.log("no auth data, skipping fetch");
      return;
    }
    if (savedEntries.length) {
      console.log("stickers already fetched");
      return;
    }
    getSettings({
      authData: auth.data,
      authType: auth.type,
    })
      .then((saved) => {
        setSavedEntries(
          saved.map(
            (s: any) =>
              ({
                url: `https://tg-sticker-bot.vercel.app/api/dev/${s.sticker}/${s.id}/video.webm`,
                emojis: [
                  // { encoded: "🍕", name: "Pizza" },
                  // { encoded: "❤️", name: "Heart" },
                ],
                name: s.id,
                stickerId: s.sticker,
                settingId: s.id,
              } as VideoEntry),
          ),
        );
      })
      .catch((err) => {
        console.error(err);
      });
  }, [auth, savedEntries.length]);

  return (
    <View padding="size-100">
      <Head>
        <title>Avoo's Sticker Stash - Telegram Sticker Templates</title>
        <meta
          name="description"
          content="Create your custom sticker pack for telegram in minutes."
        />
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main>
        <Heading level={1} UNSAFE_style={{ textAlign: "center" }}>
          <img
            src="/favicon.svg"
            alt="Logo"
            style={{
              width: "2em",
              height: "2em",
              verticalAlign: "middle",
              marginRight: "8px",
            }}
          />
          <span style={{ color: "#bf8a0f" }}>A</span>voo's{" "}
          <span style={{ color: "#bf8a0f" }}>S</span>ticker{" "}
          <span style={{ color: "#bf8a0f" }}>S</span>tash
        </Heading>
        your own
        <VideoCardList videos={savedEntries} />
        other
        <VideoCardList videos={videos} />
      </main>
    </View>
  );
}
