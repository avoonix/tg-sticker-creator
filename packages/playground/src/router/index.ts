import Vue from "vue";
import VueRouter, { RouteConfig } from "vue-router";

Vue.use(VueRouter);

const routes: Array<RouteConfig> = [
  {
    path: "/",
    name: "Editor",
    component: () =>
      import(/* webpackChunkName: "edit" */ "@/views/Editor.vue"),
  },
  {
    path: "/404",
    name: "404",
    alias: "*",
    component: () => import(/* webpackChunkName: "404" */ "@/views/404.vue"),
  },
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes,
});

export default router;
