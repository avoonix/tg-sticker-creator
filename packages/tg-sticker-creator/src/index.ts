export {
    Animatable1DValue,
    Animatable2DValue,
    AnimatableColor,
    AnimatableGradient,
    AnimatablePath,
    AnimatablePosition,
    AnimatableValue,
    AnimationError,
    BezierValues,
    BlendMode,
    CancellableCallback,
    CastError,
    ColorKeyframe,
    CompLayer,
    Coordinate,
    EachOptions,
    Easing,
    EasingArray,
    EasingName,
    easings,
    EllipseShape,
    FillRule,
    FillShape,
    FindOptions,
    GradientFillShape,
    GradientKeyframe,
    GradientStrokeShape,
    GradientType,
    GroupShape,
    ImageLayer,
    JsonArray,
    JsonObject,
    JsonValue,
    Keyframe,
    Layer,
    LayerTransform,
    LayerType,
    LineCap,
    LineJoin,
    Lottie,
    LottieItem,
    Mask,
    MaskMode,
    MatteMode,
    MergeMode,
    MergeShape,
    NullLayer,
    OffsetKeyframe,
    Path,
    PathKeyframe,
    PathShape,
    PreComp,
    RectShape,
    RepeaterShape,
    Resolvable,
    RgbaStopValue,
    RgbaValue,
    RoundedCornersShape,
    SerializationError,
    Shape,
    ShapeLayer,
    ShapeType,
    SolidLayer,
    StarShape,
    StarType,
    StrokeShape,
    TextLayer,
    ToPlainOptions,
    TransformShape,
    TrimMultiple,
    TrimShape
} from "./core/internal";
export * from "./utilities";

