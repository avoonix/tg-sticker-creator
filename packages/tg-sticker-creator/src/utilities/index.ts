import { animation } from "./animation";
import { create as c } from "./create";
export { BlinkAnimationOptions, ParticleAnimationOptions, RandomizableValue } from "./animation";
export * from "./enum";
export { optimizeFilesize, OptimizeFilesizeOptions } from "./optimize";

export const create = {
  ...c,
  animation,
};
