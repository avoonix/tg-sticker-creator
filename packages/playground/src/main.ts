import "firacode";
import "reflect-metadata";
import Vue from "vue";
import App from "./App/App.vue";
import { registerGlobalErrorHandlers } from "./Notifications/globalHandler";
import vuetify from "./plugins/vuetify";
import router from "./router";

registerGlobalErrorHandlers();

Vue.config.productionTip = false;

new Vue({
  router,
  vuetify,
  render: (h) => h(App),
}).$mount("#app");
