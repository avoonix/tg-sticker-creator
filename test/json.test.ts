import fs from "fs";
import path from "path";
import snapshotDiff from "snapshot-diff";
import { create } from "../src";

const sticker1 = fs.readFileSync(path.join(__dirname, "sticker1.json"), {
  encoding: "utf-8",
});

describe("json", () => {
  it("can stringify sticker multiple times", () => {
    const animation = create.stickerFromTgs(sticker1);
    const plain1 = animation.toPlain({ precision: Infinity });
    const plain2 = animation.toPlain({ precision: Infinity });
    expect(snapshotDiff(plain1, plain2)).toMatchSnapshot();
  });
  it("can parse sticker multiple times", () => {
    const first = create.stickerFromTgs(sticker1);
    const second = create.stickerFromTgs(sticker1);
    expect(snapshotDiff(first, second)).toMatchSnapshot();
  });
  it("can stringify and parse sticker multiple times", () => {
    const parsedOnce = create.stickerFromTgs(sticker1);
    const parsedTwice = create.stickerFromTgs(
      create.stickerFromTgs(sticker1).toTgsString({ precision: Infinity })
    );
    expect(snapshotDiff(parsedOnce, parsedTwice)).toMatchSnapshot();
  });
  it("imports correctly", () => {
    const animation = create.stickerFromTgs(sticker1);
    expect(
      snapshotDiff(
        JSON.parse(sticker1),
        JSON.parse(JSON.stringify(animation.toPlain({ precision: Infinity })))
      )
    ).toMatchSnapshot();
  });
});
