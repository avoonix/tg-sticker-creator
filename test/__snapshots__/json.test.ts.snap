// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json can parse sticker multiple times 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`json can stringify and parse sticker multiple times 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`json can stringify sticker multiple times 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`json imports correctly 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -46,11 +46,10 @@
          },
          \\"sk\\": Object {
            \\"a\\": 0,
            \\"k\\": 0,
          },
-         \\"ty\\": \\"tr\\",
        },
        \\"op\\": 180,
        \\"shapes\\": Array [
          Object {
            \\"bm\\": 0,
@@ -12685,11 +12684,10 @@
          },
          \\"sk\\": Object {
            \\"a\\": 0,
            \\"k\\": 0,
          },
-         \\"ty\\": \\"tr\\",
        },
        \\"op\\": 180,
        \\"shapes\\": Array [
          Object {
            \\"bm\\": 0,
@@ -25353,11 +25351,10 @@
          },
          \\"sk\\": Object {
            \\"a\\": 0,
            \\"k\\": 0,
          },
-         \\"ty\\": \\"tr\\",
        },
        \\"op\\": 180,
        \\"shapes\\": Array [
          Object {
            \\"bm\\": 0,
@@ -37992,11 +37989,10 @@
          },
          \\"sk\\": Object {
            \\"a\\": 0,
            \\"k\\": 0,
          },
-         \\"ty\\": \\"tr\\",
        },
        \\"op\\": 180,
        \\"shapes\\": Array [
          Object {
            \\"bm\\": 0,"
`;
